pkg/signatures/engine/engine.go

```diff
@@ -245,7 +245,7 @@ func (engine *Engine) loadSignature(signature detect.Signature) (string, error)
 		return "", fmt.Errorf("failed to store signature: signature \"%s\" already loaded", metadata.Name)
 	}
 	engine.signaturesMutex.RUnlock()
-	if err := signature.Init(engine.matchHandler); err != nil {
+	if err := signature.Init(detect.SignatureContext{Callback: engine.matchHandler, Logger: logger.Current()}); err != nil {
```

Comment:
Couldn't the signature package just import the logger and use it at the package level?

Reply:
Ah never-mind. I got the need.